arquin/device/main.py

```diff
@@ -0,0 +1,41 @@
+import networkx as nx
+import numpy as np
+
+from arquin.module.main import Module
+
+class Device:
+    def __init__(self, device_graph, module_graphs):
+        '''
+        device_graph: graph of the device. Each module is a node. Inter-module connections are the edges.
```

Comment:
Not enough information is being provided here, which is why qubits are being randomly assigned in `_connect_modules` below. Whenever a `Device` is created we always need to know 
1. Which modules are connected, and
2. Which qubits of those modules are connected
We may as well provide this information in `__init__` to be specific and avoid randomly assigning qubits in the `global_edges`

Reply:
This is left as random for now. After we are done developing the compiler itself we can add specification of which qubits to connect.

Reply:
It doesn't make sense to me to leave this until after the compiler is finished. How we define the `Device` seems pretty crucial. I've updated `__init__` and will push those changes here after I resolve the merge conflicts from your latest push

Reply:
Is this PR still tracking changes on my branch even after I created it?